### DISCUSSION TITLE
Clear the query not reload the page.

### DISCUSSION THREAD

**Discussion by shalldie** [vercel/next.js#68021] [Help] [answered]
### Summary

Next14, app router.
I've a page, use a query in url: `/article?label=xxx`.

When query label changes, it can research corrently.
But when route change from `/article?label=xxx` to `/article`, page not reload.

It seem to hit the client cache, how can it reload page when the query is cleared?

### Additional information

Update: 
here's the [codesandbox](https://codesandbox.io/p/devbox/bold-smoke-px769f)

steps:
1. go to '/article?label=two'
2. click refresh in navbar.
3. click [clear filter] to '/article'
4. the info in page is still old, unless click refresh.

my next version is `14.2.4`, and I find that `14.1.x` works fine.

**Comment by little077** [comment]
I didn't understand what you meant. Can you reproduce it using a tool similar to a codebox? And inform you of your ideal expected behavior

**Reply by shalldie** [reply]
I add a codesandbox in Additional information

**Reply by icyJoseph** [reply]
You didn't save the correct link to the sandbox, use Stackblitz instead, it is way better, just type: `next.new` into your browser URL. Don't forget to save the changes you make, and just share the link.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Could it be this bug? 

- https://github.com/vercel/next.js/issues/64170
- https://github.com/vercel/next.js/issues/65030

**It has a clear demonstration, https://nextjs-queryparams.vercel.app/  and it has been _fixed_ in canary.** 

Alternatively to fix it on the current release, one could try to use https://nextjs.org/docs/app/api-reference/next-config-js/staleTimes, but it is not ideal at all.

**Reply by shalldie** [reply]
I update the Additional information and add  codesandbox link.
It seems to be like the issue you provided, that's it !

---
Thank you for the pacient, I'll learn more about codesandbox and stackblitz later.
